doc/classes/Object.xml

```diff
@@ -315,7 +315,7 @@
 			<return type="Variant" />
 			<param index="0" name="method" type="StringName" />
 			<description>
-				Calls the [param method] on the object during idle time. This method supports a variable number of arguments, so parameters can be passed as a comma separated list.
+				Calls the [param method] on the object during idle time, which mainly happens at the end of process and physics frames. Idle time will run deferred calls until there are none left, so even if this is used during idle time itself, the function will still run in the current idle time cycle. This method supports a variable number of arguments, so parameters can be passed as a comma separated list.
```

Comment:
The description already implies it, but using `call_deferred()` inside deferred call will just put another call to the current queue, resulting in potentially infinite chain of calls. Maybe we could warn about it more explicitly?

Also you *can* queue deferred call inside deferred call, so it might be worth it to put a code sample.
```GDScript
get_tree().process_frame.connect(my_method, CONNECT_ONE_SHOT)
```
to simply call it later and
```GDScript
get_tree().process_frame.connect(my_method, CONNECT_ONE_SHOT | CONNECT_DEFERRED)
```
to wait a full frame.

Reply:
> Maybe we could warn about it more explicitly?

Hmm, I tried to convey that as well as I could without making the description too long for something that won't matter in most usecases.
If length doesn't matter, then maybe the clearest explanation would be an explicit example, something like:
```gdscript
# This function will freeze the game, as it endleslly adds itself to the deferred queue
# The queue never empties, so the game's processing can't continue
func infinite_deferred_calls():
    call_deferred("infinite_deferred_calls")
```
I'm not sure if adding a "negative example" breaks some sort of rule for good documentation.

> Also you _can_ queue deferred call inside deferred call

I was gonna say the main action there is "waiting for the next frame through signals" and not "deferring the call", but now that I'm thinking about it, it could work. And "waiting for a frame" might be what some people look for when they look up deferred calls